crates/red_knot_project/src/metadata/configuration_file.rs

```diff
@@ -0,0 +1,69 @@
+use std::sync::Arc;
+
+use ruff_db::system::{System, SystemPath, SystemPathBuf};
+use thiserror::Error;
+
+use crate::metadata::value::ValueSource;
+
+use super::options::{KnotTomlError, Options};
+
+/// A `knot.toml` configuration file with the options it contains.
```

Comment:
This mentions `knot.toml` explicitly because this struct does not apply to `pyproject.toml` files?

Reply:
Yes, we don't pick up `pyproject.toml` in user directories (unlike ruff, the same as uv) because `pyproject.toml`s are for projects